src/core/useMutableSource.ts

```diff
@@ -49,6 +48,20 @@ export const useMutableSource = (
       /* [3] */ currentVersion,
       /* [4] */ currentSnapshot,
     ])
+  } else if (
```

Comment:
@Aslemammad This is more correct fix. But, it might not be important for the jotai usage.

Reply:
No problem, it's great.